### DISCUSSION TITLE
Quick start guide

### DISCUSSION THREAD

**Discussion by longshadowdev** [nextras/orm#576] [General]
Hello, this ORM looks nice. So far I've been able to get the Dbal part set up and working. I've also used the Contributte ORM generator to reverse engineer my database into models/collections etc.

I've read the documentation for Orm. It seems to jump from a demonstration of how to do selects/inserts, but no actual setup guide? For example, how do I actually instantiate the 'Orm' class that extends Nextras\Orm\Model\Model? I would expect I need to pass in my database credentials or an instance of Nextras\Dbal\Connection.

I know there's an example using some Nette dependency injection. Is that mandatory? I'm currently using ThePhpLeague container for DI in my project.

Is this tightly coupled to Nette DI and whatever neon is and there's no way to use it by itself?

**Comment by hrach** [comment]
Currently, there is only "free" setup for Nette DI. We plan to support Symfony DI, [PR is here](https://github.com/nextras/orm/pull/490). Similarly, you can implement something like that for ThePhpLeague container.

There is also something like "simple" solution: [SimpleModelFactory.php](https://github.com/nextras/orm/blob/d2a1cd5b8740f62a94fc2dd0c0a78346aa4cf7e0/src/Model/SimpleModelFactory.php), its usage can be found in [tests](https://github.com/nextras/orm/blob/d2a1cd5b8740f62a94fc2dd0c0a78346aa4cf7e0/tests/cases/integration/Mapper/file.general.phpt) (This one is not using Dbal, but it may shed some light how to setup, your mappers will require the connection you are already able to construct). This may be a quick start for you before implementing the DI solution.

**Comment by longshadowdev** [comment]
Thank you @hrach, I have been playing around and did get this working using my own DI. I will clean up what I've done and then reply here again some sample code so that others can see.
